express-notes-api/index.js

```diff
@@ -0,0 +1,94 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+const notes = require('./data.json');
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const resArr = [];
+  for (const prop in notes.notes) {
+    const resJSON = notes.notes[prop];
+    resArr.push(resJSON);
+  }
+  res.status(200).json(resArr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  if (Number.isInteger(parseInt(req.params.id)) && parseInt(req.params.id) > 0) {
```

Comment:
Forcing `req.params.id` to an integer with `parseInt` before checking if it's an integer won't catch non-whole numbers.

Reply:
Gotcha. 
![image](https://user-images.githubusercontent.com/75236087/130553891-088f3031-9428-47e6-9985-a25c5fdc17ed.png)